build/tools/Automation/Python/Python.cs

```diff
@@ -0,0 +1,159 @@
+            {
+                // Deserialize constants.yaml
+                string constantsYamlAbsolutePath = Path.Combine(oryxRootPath, "build", Constants.ConstantsYaml);
+                List<YamlConstants> yamlConstantsObjs = await this.yamlFileReaderService.ReadConstantsYamlFileAsync(constantsYamlAbsolutePath);
+
+                this.UpdateOryxConstantsForNewVersions(versionObjs, yamlConstantsObjs, oryxRootPath);
+            }
+        }
+
+        public async Task<List<VersionObj>> GetNewVersionObjsAsync()
+        {
+            var url = "https://www.python.org/api/v2/downloads/release/";
+            var response = await this.httpClient.GetDataAsync(url);
+            var releases = JsonConvert.DeserializeObject<List<Models.Release>>(response);
+
+            url = "https://oryx-cdn.microsoft.io/python?restype=container&comp=list&include=metadata";
+            HashSet<string> oryxSdkVersions = await this.httpClient.GetOryxSdkVersionsAsync(url);
+
+            var versionObjs = new List<VersionObj>();
+            foreach (var release in releases)
+            {
+                string newVersion = release.Name.Replace("Python", string.Empty).Trim();
+
+                if (!release.PreRelease &&
+                    this.versionService.IsVersionWithinRange(newVersion, minVersion: "3.10.9") &&
+                    !oryxSdkVersions.Contains(newVersion))
+                {
+                    Console.WriteLine($"newVersion {newVersion}");
+                    versionObjs.Add(new VersionObj
+                    {
+                        Version = newVersion,
+                        GpgKey = this.GetGpgKeyForVersion(newVersion),
+                    });
+                }
+            }
+
+            return versionObjs;
+        }
+
+        private void UpdateOryxConstantsForNewVersions(List<VersionObj> versionObjs, List<YamlConstants> yamlConstants, string oryxRootPath)
+        {
+            Dictionary<string, YamlConstants> pythonYamlConstants = this.GetYamlPythonConstants(yamlConstants);
+
+            foreach (var versionObj in versionObjs)
+            {
+                string version = versionObj.Version;
+                string pythonConstantKey = this.GeneratePythonConstantKey(versionObj);
+                Console.WriteLine($"[UpdateConstants] version: {version} pythonConstantKey: {pythonConstantKey}");
+                pythonYamlConstants["python-versions"].Constants[pythonConstantKey] = version;
+
+                this.UpdateVersionsToBuildTxt(versionObj, oryxRootPath);
+            }
+
+            var serializer = new SerializerBuilder()
+                .WithNamingConvention(UnderscoredNamingConvention.Instance)
+                .Build();
+
+            var constantsYamlAbsolutePath = Path.Combine(oryxRootPath, "build", Constants.ConstantsYaml);
+            var stringResult = serializer.Serialize(yamlConstants);
+            File.WriteAllText(constantsYamlAbsolutePath, stringResult);
```

Comment:
I would expect all of this functionality to occur within the `IYamlFileReaderService` interface, although it may then need to be changed to something like `IYamlFileService` so it handles _both_ reading and writing

Reply:
Yup! I've update to YamlFileService to include both read and write